osfoffline/sync/local.py

```diff
@@ -21,11 +22,19 @@ class LocalSyncWorker(ConsolidatedEventHandler, metaclass=Singleton):
 
     def __init__(self):
         super().__init__()
-        self.folder = get_current_user().folder
+        try:
+            user = get_current_user()
+        except NoResultFound:
+            user = None
+            self.folder = None
+
+        if user:
+            self.folder = user.folder
 
         self.observer = Observer()
         self.ignore = threading.Event()
-        self.observer.schedule(self, self.folder, recursive=True)
+        if self.folder:
```

Comment:
Can you help me understand this a bit better? It appears that if there is no user, it will never schedule an observer.

But:
1. Under what conditions does a `LocalSyncWorker` get scheduled if there is no logged-in user? It seems like this code shouldn't even be reachable if line 27 throws an exception.
2. I don't see a mechanism here for the observer to get scheduled once a user does become available. (and since this is a singleton, nothing short of logging out would lead to the creation of a new instance)

Does this line create a situation where watchdog simply wouldn't run for the user at all?